core/domain/takeout_service.py

```diff
@@ -66,5 +67,28 @@ def export_data_for_user(user_id):
         final_name = ('_').join([x.lower() for x in split_name]) + '_data'
         exported_data[final_name] = model.export_data(user_id)
 
-    # Combine the data into a single dictionary.
-    return exported_data
+    # Separate out images. We store the images that need to be separated here
+    # as a dictionary mapping tuples to strings. The tuple value indicates the
+    # "path" to take to the image in the user's data dictionary, and the string
+    # indicates the filename that the exported image will be saved to.
+    image_positions = {
+        ('user_settings_data', 'profile_picture_data_url'):
+            'user_settings_profile_picture.png'
+    }
+    image_files = []
+    for image_position in image_positions:
+        # Move pointer to the position indicated by the tuple.
+        pointer = exported_data
+        for key in image_position[:-1]:
+            pointer = pointer[key]
+
+        # Swap out data with replacement filename.
+        image_key = image_position[-1]
+        image_data = pointer[image_key]
+        if pointer[image_key] != None:
+            replacement_filename = image_positions[image_position]
+            image_files.append(
+                takeout_domain.TakeoutImage(image_data, replacement_filename))
+            pointer[image_key] = replacement_filename
```

Comment:
Add a comment to explain what this is doing.

Should the keyname be changed (since what its pointing to is now a different thing -- it's not a data url any more)?

Reply:
Sure I can do this, but then I need to store the information about the replacement key too, which I would have to store in the dictionary image_positions. Meaning I would have a mapping from tuples to another domain object representing the new key and value for this image. Is this the approach you want me to take?

Reply:
Mmm. Yeah I think so, that makes sense!

Reply:
Done